sympy/calculus/util.py

```diff
@@ -112,3 +119,601 @@ def elm_domain(expr, intrvl):
+            return Add(self, -other, evaluate=False)
+        return NotImplemented
+
+    @_sympifyit('other', NotImplemented)
+    def __rsub__(self, other):
+        return self.__neg__() + other
+
+    @_sympifyit('other', NotImplemented)
+    def __mul__(self, other):
+        if isinstance(other, Expr):
+            if isinstance(other, Limits):
+                return Limits(Min(Mul(self.min, other.min),
+                            Mul(self.min, other.max),
+                            Mul(self.max, other.min),
+                            Mul(self.max, other.max)),
+                        Max(Mul(self.min, other.min),
+                            Mul(self.min, other.max),
+                            Mul(self.max, other.min),
+                            Mul(self.max, other.max)))
+            if other is S.Infinity:
+                if self.min.is_zero:
+                    return Limits(0, oo)
+                if self.max.is_zero:
+                    return Limits(-oo, 0)
+            if other is S.NegativeInfinity:
+                if self.min.is_zero:
+                    return Limits(-oo, 0)
+                if self.max.is_zero:
+                    return Limits(0, oo)
+            if other.is_real:
+                if other.is_zero:
+                    if self == Limits(-oo, oo):
+                        return Limits(-oo, oo)
+                    if self.max is S.Infinity:
+                        return Limits(0, oo)
+                    if self.min is S.NegativeInfinity:
+                        return Limits(-oo, 0)
+                    return S.Zero
+                if other.is_positive:
+                    return Limits(Mul(self.min, other), Mul(self.max, other))
+                elif other.is_negative:
+                    return Limits(Mul(self.max, other), Mul(self.min, other))
+            if isinstance(other, Order):
+                return other
+            return Mul(self, other, evaluate=False)
+        return NotImplemented
+
+    __rmul__ = __mul__
+
+    @_sympifyit('other', NotImplemented)
+    def __div__(self, other):
+        if isinstance(other, Expr):
+            if isinstance(other, Limits):
+                if not S.Zero in other:
+                    return self*Limits(1/other.max, 1/other.min)
+
+                if S.Zero in self and S.Zero in other:
+                    if self.min.is_zero and other.min.is_zero:
+                        return Limits(0, oo)
+                    if self.max.is_zero and other.min.is_zero:
+                        return Limits(-oo, 0)
+                    return Limits(-oo, oo)
+
+                if self.max.is_negative:
+                    if other.min.is_negative:
+                        if other.max.is_zero:
+                            return Limits(self.max/other.min, oo)
+                        if other.max.is_positive:
+                            return Union(Limits(-oo, self.max/other.max),
```

Comment:
This should probably be just ``Limits(-oo, oo)`` for now. Maybe something better could be invented later.